src/main/sarl/fr/utbm/info/vi51/worldswar/environment/Environment.sarl

```diff
@@ -202,22 +207,19 @@ agent Environment {
 	 */
 	on InfluenceEvent {
 		synchronized(this) {
-			var body = agentBodies.get(occurrence.uuid)
-			if(body === null) {
-				warning("Influence received from agent #" + occurrence.uuid + ", but this agent does not exist in the current Environment.")
-			}
-			
-			if(body.influence !== null) {
-				error("Agent #" + occurrence.uuid + " has sent multiple influences in the same step")
-			}
-			else {
-				body.influence = occurrence.influence
-				influencesReceived++
-				
-				if(influencesReceived == agentBodies.size) {
-					applyStep
+			for(body : this.agentBodies.values){
```

Comment:
Ce n'est sans doute plus utile de copier les influences dans les corps, si on les a sous forme de hashMap.
Cette partie pourrait sans doute être supprimée, et applyAntInfluence pourrait être basée sur la hashMap ?

BTW, le message "has sent multiple influences in the same step" n'est pas à jour